cmd/kube-proxy/app/server.go

```diff
@@ -751,14 +750,14 @@ func (s *ProxyServer) Run() error {
 	serviceConfig.RegisterEventHandler(s.Proxier)
 	go serviceConfig.Run(wait.NeverStop)
 
-	if s.UseEndpointSlices {
+	if endpointsProxier, ok := s.Proxier.(config.EndpointsHandler); ok {
```

Comment:
this gives preference to endpoints, shouldn't we keep using endpointslices by default if they are present?

Reply:
I mentioned that in the initial comment; if we wanted this to be "if it implements EndpointSliceHandler then use EndpointSlices", then that that would have to imply that `EndpointSliceHandler` was not a required part of the `proxy.Provider` implementation.

Hm... maybe we could test if the provider can be cast to `config.NoopEndpointSliceHandler` ?

Reply:
> Hm... maybe we could test if the provider can be cast to `config.NoopEndpointSliceHandler` ?

no, that doesn't work. I pushed a new commit with a slightly modified version of the original... which is still kind of weird, but maybe better?

Reply:
```
if s.ProxyMode == ProxyModeUserspace {
 // use endpoints
} else {
 // use slices
}
```
my concern with casting is that seems easy to have a mistake and start using Endpoints